### PR TITLE
Removed mt5 dependency on t5

### PR DESCRIPTION
**Pull request by harry7337**
# What does this PR do?

Fixes #19303. Removes the dependency of mt5 on t5.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.
@sgugger

### CODE REVIEW

**Comment by harry7337** [comment]
Creating this PR after a long time because of a tensorflow AVX problem on my system due to which I couldn't run any tests. Currently working on my friend's laptop and making changes:)

File: src/transformers/models/mt5/modeling_mt5.py [hunk 1]
```diff
@@ -14,8 +14,26 @@
...
+from ...utils.model_parallel_utils import assert_device_map, get_device_map
+from ..t5.configuration_t5 import T5Config
+from ..t5.modeling_t5 import (
+    DEPARALLELIZE_DOCSTRING,
+    PARALLELIZE_DOCSTRING,
+    T5_ENCODER_INPUTS_DOCSTRING,
+    T5_INPUTS_DOCSTRING,
+    T5PreTrainedModel,
+    T5Stack,
+)
```
Comment (sgugger) [inline src/transformers/models/mt5/modeling_mt5.py:36]: The goal is to not have those imports but copy those objects here with appropriate `# Copied from` statements.

File: src/transformers/models/mt5/modeling_tf_mt5.py [hunk 2]
```diff
@@ -14,8 +14,31 @@
...
+from ...tf_utils import shape_list
+from ...utils import (
+    DUMMY_INPUTS,
+    DUMMY_MASK,
+    add_start_docstrings_to_model_forward,
+    logging,
+    replace_return_docstrings,
+)
+from ..t5.configuration_t5 import T5Config
+from ..t5.modeling_tf_t5 import _HEAD_MASK_WARNING_MSG, T5_INPUTS_DOCSTRING, TFT5MainLayer
```
Comment (sgugger) [inline src/transformers/models/mt5/modeling_tf_mt5.py:41]: Same here.

**Review by sgugger — COMMENTED** [review]
Thanks for working on this! We need to push it one step further though, the modeling files for mT5 shouldn't use any of the objects of T5!

**Pull request closed without merging**